rust/azure_iot_operations_services/src/azure_device_registry/models/asset.rs

```diff
@@ -89,6 +89,54 @@ pub struct AssetSpecification {
     pub version: Option<u64>,
 }
 
+/// Represents the partial specification of an Discovered Asset in the Azure Device Registry service.
+pub struct DiscoveredAssetSpecification {
+    /// URI or type definition ids.
+    pub asset_type_refs: Vec<String>, // if None, we can represent as empty vec. Can currently only be length of 1
```

Comment:
> Can currently only be length of 1

This was okay to have just in the documentation for received structs, but since this is a struct that will be sent, we should consider adding validation for this. And for all other fields like this. Check the discovered_asset.yaml to find these restrictions (or confirm with Abhipsa if this is something that's fine for the edge to send and then the operator would filter out)

Reply:
Did a quick look and not all fields that can only be length of 1 on the non discovered flow have the same limit for the discovered flow, fyi

Reply:
I"ll just remove it wholesale - I'm not trying to validate this right now.